openssl/src/asn1.rs

```diff
@@ -480,6 +557,40 @@ impl fmt::Debug for Asn1StringRef {
     }
 }
 
+impl FromAsn1Type<Asn1StringRef> for Asn1StringRef {
+    fn from_asn1type(ty: &Asn1TypeRef) -> Result<&Asn1StringRef, Asn1Error> {
+        unsafe {
+            unsafe fn from_asn1type_ptr(ty: &Asn1TypeRef) -> &Asn1StringRef {
+                Asn1StringRef::from_const_ptr(
+                    (*ty.as_ptr()).value.asn1_string as *const ffi::ASN1_STRING,
+                )
+            }
+            match ty.typ()? {
+                Asn1TagValue::BitString => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::BmpString => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::Enumerated => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::GeneralString => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::GeneralizedTime => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::GraphicString => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::Ia5String => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::Integer => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::NumericString => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::OctetString => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::PrintableString => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::T61String => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::UniversalString => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::UtcTime => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::Utf8String => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::VideotexString => Ok(from_asn1type_ptr(ty)),
+                Asn1TagValue::VisibleString => Ok(from_asn1type_ptr(ty)),
+                _ => Err(Asn1Error {
+                    message: String::from("Not a string type. Conversion not supported."),
```

Comment:
I'd rather this just return an option than make a new error type.

Reply:
Fixed by turning `Result` to `Option`. But we don't report error reasons this way...